File: Support Vector Regression/SVR.py
# SVR regression

# Import the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# import the dataset
dataset = pd.read_csv("data/Position_Salaries.csv")

# remove the first column
X = dataset.iloc[:, 1:2].values # matrix of features
y = dataset.iloc[:, 2].values # dependent variable

# splitting data in train and test sets - not needed for this exercise as datas et is too small
# library(caTools)
# set.seed(123)
# split <- caTools::sample.split(dataset$Purchased, SplitRatio = 0.8) # returns 80% TRUE and 20% FALSE
# training_set <- subset(dataset, split == TRUE) # Take TRUE values
# test_set <- subset(dataset, split == FALSE) # Take false values
#
#feature scaling 
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(np.array(X).reshape(1, -1))
y = sc_y.fit_transform(y)


# Fit SVR to the dataset
# Create your regressor
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf') # rbf is a non-linear kernel
regressor.fit(X, y)

# Predict new results 
y_pred = regressor.predict(np.array(6.5).reshape(1, -1))

# Visualizing the Regression Model results (for higher resolution and smoother curve)
plt.scatter(X, y, color = 'red')
plt.plot(X, regressor.predict(X), color = 'blue')
plt.title('Truth or Bluff (SVR)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
                                       

File: Support Vector Regression/SVR.R
# SVR
library(e1071)

# Importing the data set
dataset <- read.csv("data/position_salaries.csv")

# remove the first column
dataset <- dataset[,2:3]

# splitting data in train and test sets - not needed for this exercise as data s \et is too small
# library(caTools)
# set.seed(123)

# split <- caTools::sample.split(dataset$Purchased, SplitRatio = 0.8) # returns 80% TRUE and 20% FALSE
# training_set <- subset(dataset, split == TRUE) # Take TRUE values
# test_set <- subset(dataset, split == FALSE) # Take false values
# 
# # feature scaling - need for this either
# training_set[,2:3] <- scale(training_set[,2:3])
# test_set[,2:3] <- scale(test_set[,2:3])

# Fitting the Regression  Model to the data set
# Create your regressor
regressor <- svm(formula = Salary ~ .,
                 data = dataset,
                 type = "eps-regression") # be sure to fill out type. Default kernel is Gaussian

# Predicting a new result 
# create a new data set to examine a single data point
y_pred <- predict(regressor, data.frame(Level = 6.5))

# Visualizing the Regression Model results (for higher resolution and smoother curve)
library(ggplot2)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), # real observation points
             colour = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)), # predicted observation points
            colour = 'blue') +
  ggtitle("Truth or Bluff (SVR)") + 
  xlab('') +
  ylab('') +
  theme_bw()
